Userland/Libraries/LibWeb/Layout/Node.cpp

```diff
@@ -253,7 +253,14 @@ bool Node::is_floating() const
 
 bool Node::is_positioned() const
 {
-    return has_style() && computed_values().position() != CSS::Position::Static;
+    if (!has_style())
+        return false;
+    if (is_grid_item() && computed_values().z_index().has_value()) {
+        // https://www.w3.org/TR/css-grid-2/#z-order
+        // grid items with z_index should behave as if position were "relative"
+        return true;
+    }
```

Comment:
This feels incomplete, as we also have code that checks `position() == CSS::Position::Relative` in multiple places.
To make this fool-proof, maybe we can override the `position()` value for grid items so all code always does the right thing?

Reply:
on a second look, turned out that instead grid items should behave as "relative" with z-index != auto only for painting purposes. for example "top", "right", "bottom", "left" properties should not work if grid item has position != auto. so instead I introduced `should_be_treated_as_positioned_for_painting()` that is used only in painting.